### PR TITLE
Add map and Typescript

### PR DESCRIPTION
**Pull request by sonnylazuardi**
This PR will add
- Typescript.
- Mapbox in `/map` page.

Please add Mapbox token from https://account.mapbox.com/access-tokens/ as env variable

```
NEXT_PUBLIC_MAPBOX_KEY=YOUR_MAPBOX_KEY
```

Result: https://jakarta-vax-availability-theta.vercel.app/map

https://user-images.githubusercontent.com/856609/124431677-3e538200-dda3-11eb-9cce-44dc43350a3e.mp4

### CODE REVIEW

**Comment by angeloanan** [comment]
Some results from the API returns an empty `detail_lokasi` field resulting in some vaccination sites not displayed in the map.

Should we handle this? (Relates to #8)

**Comment by sonnylazuardi** [comment]
> Some results from the API returns an empty `detail_lokasi` field resulting in some vaccination sites not displayed in the map.
> 
> Should we handle this? (Relates to #8)

I handle this in filter function, is it not enough?

https://github.com/tibudiyanto/jakarta-vax-availability/pull/9/files#diff-696d52574dda2a3148dad50df0b171a8b883aacbe8f5e794c513d9ae16481530R85

**Comment by sonnylazuardi** [comment]
ah you mean how to display it if it isn't in the map? then i think need to go back and use the listing page (main page).

**Comment by tibudiyanto** [comment]
Amazing. What do i need to do to deploy these? Only API key? @sonnylazuardi

**Comment by sonnylazuardi** [comment]
> Amazing. What do i need to do to deploy these? Only API key? @sonnylazuardi

Yes, only API key from https://account.mapbox.com/access-tokens/

env name: `NEXT_PUBLIC_MAPBOX_KEY`

File: next.config.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require('dotenv').config()
```
Comment (pveyes) [inline next.config.js]: I thought next.js already have a way to load env var automatically, without changes in next.config?

https://nextjs.org/docs/basic-features/environment-variables#exposing-environment-variables-to-the-browser

Comment (tibudiyanto) [reply next.config.js]: Yep was going to comment about this too. do you mind removing this? @sonnylazuardi

Comment (sonnylazuardi) [reply next.config.js]: ah correct 💯, removed dotenv package

Comment (tibudiyanto) [reply next.config.js]: thanks

**Comment by tibudiyanto** [comment]
I am not that familiar with typescript. @pveyes if you say its gucci imma hit the merge.

File: next.config.js (hunk 1 above)
Comment (dionaditya) [reply next.config.js]: What about use pigeon-maps instead raw mapbox ? No need api key
https://github.com/mariusandra/pigeon-maps

File: package.json [hunk 2]
```diff
@@ -13,9 +13,14 @@
...
     "next": "latest",
     "react": "^17.0.2",
-    "react-dom": "^17.0.2"
+    "react-dom": "^17.0.2",
+    "react-mapbox-gl": "^5.1.1"
   },
-  "license": "MIT"
+  "license": "MIT",
+  "devDependencies": {
+    "typescript": "^4.3.5"
```
Comment (pveyes) [inline package.json:24]: Wait so `@types/react` is optional? TIL

**Review by pveyes — APPROVED** [review]
LGTM if CI green

**Pull request merged**